src/Analyzers/CSharp/Analyzers/NewLines/EmbeddedStatementPlacement/EmbeddedStatementPlacementDiagnosticAnalyzer.cs

```diff
@@ -59,6 +62,9 @@ private void Recurse(SyntaxTreeAnalysisContext context, ReportDiagnostic severit
 
             foreach (var child in node.ChildNodesAndTokens())
             {
+                if (context.FilterSpan.HasValue && !child.Span.IntersectsWith(context.FilterSpan.GetValueOrDefault()))
+                    continue;
```

Comment:
for the loops consider passing in the filter-span (which i would consider making non-nullable).  Then it's just a simple intersects call at each layer.  When there is no filter-span, just pick the entire file as the filterspan.

Actually... would that just make things simpler, if FilterSpan was non-nullable, and was just given the full file span when we're etting diags for the file?

Reply:
> Actually... would that just make things simpler, if FilterSpan was non-nullable, and was just given the full file span when we're etting diags for the file?

I personally feel it is better to retain `FilterSpan` as a nullable span - it makes it explicit if we are analyzing the whole tree or a sub-span within the tree. Analyzers can easily make a call of whether to execute certain special filtering logic only for the span case, and avoid it for full tree case.